internal/provider/resource_metadata.go

```diff
@@ -70,9 +71,22 @@ var metadataTypes = []string{
 	"PagerDutyWebhook",
 }
 
+const scalarMetadataTypesCount = 1
```

Comment:
I really just wanted to make sure that we always remember to update metadata types everywhere.

The `strings.Join(metadataTypes[:scalarMetadataTypesCount], "`, `")` is a bit harder to read, but I don't mind since you see the generated docs as well. WDYT?